frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { DashboardCardHelp } from '@console/internal/components/dashboard/dashboard-card';
+import {
+  humanizeBinaryBytesWithoutB,
+  humanizePercentage,
+} from '@console/internal/components/utils';
+
+const EfficiencyRow: React.FC<RowProps> = React.memo(({ value, infoText }) => {
+  return (
+    <div className="nb-object-data-reduction-card__row-status-item">
+      {_.isNil(value) ? (
+        <span className="text-secondary">Unavailable</span>
+      ) : (
+        <span className="nb-object-data-reduction-card__row-status-item-text">
+          {Number(value).toFixed(1)}:1
+        </span>
+      )}
+      <DashboardCardHelp>{infoText}</DashboardCardHelp>
+    </div>
+  );
+});
+
+const SavingsRow: React.FC<RowProps> = React.memo(({ value, infoText, logicalSize }) => {
+  let text = <span className="text-secondary">Unavailable</span>;
+  if (!_.isNil(value)) {
+    const savingsPercentage = humanizePercentage((100 * Number(value)) / Number(logicalSize));
+    const savings = humanizeBinaryBytesWithoutB(Number(value));
+    text = (
+      <span className="nb-object-data-reduction-card__row-status-item-text">
+        {savings.value}
+        {savings.unit}({savingsPercentage.string})
+      </span>
+    );
+  }
+  return (
+    <div className="nb-object-data-reduction-card__row-status-item">
+      {text}
+      <DashboardCardHelp>{infoText}</DashboardCardHelp>
+    </div>
+  );
+});
+
+export const ObjectDataReductionItem: React.FC<ObjectDataReductionItemProps> = React.memo(
+  ({ title, value, infoText, logicalSize }) => (
+    <div className="nb-object-data-reduction-card__row">
+      <div className="nb-object-data-reduction-card__row-title">{title}</div>
```

Comment:
To be more transparent, i'd suggest to use `SavingsItem` and `EfficiencyItem` directly and reuse `ObjectDataReductionItem` within them like

```
export const ObjectDataReductionItem: React.FC<ObjectDataReductionItemProps> = React.memo(
  ({ title, infoText, children }) => (
    <div className="nb-object-data-reduction-card__row">
      <div className="nb-object-data-reduction-card__row-title">{title}</div>
        <div className="nb-object-data-reduction-card__row-status-item">
          {children}
          <DashboardCardHelp>{infoText}</DashboardCardHelp>
        </div>
    </div>
  </div>
));
```

and inside `SavingsItem`

```
return (
  <ObjectDataReductionItem title={...} infoText={...}>
      {text}
  </ObjectDataReductionItem>
);
```

same for `EfficiencyItem`

Reply:
@rawagner thanx ...looks cool now!!